### PR TITLE
#74 provide integer (or binary) based EnumSet

### PR DESCRIPTION
**Pull request by marc-mabe**
see #74

### CODE REVIEW

File: src/EnumSet.php [hunk 1]
```diff
@@ -484,36 +690,132 @@ public function setBinaryBitsetBe($bitset)
...
+     *
+     * @param int $ordinal Ordnal number of bit to set
+     * @param bool $bit    The bit to set
+     * @return void
+     * @see doSetBitBin
+     * @see doSetBitInt
+     * @see doUnsetBin
+     * @see doUnsetInt
+     */
+    public function setBit($ordinal, $bit)
```
Comment (prolic) [inline src/EnumSet.php]: when no PHP 7 is in use, maybe we should assert that $bit is a boolean

Comment (prolic) [inline src/EnumSet.php]: maybe same for $ordinal, assert that it is integer

File: src/EnumSet.php [hunk 2]
```diff
@@ -484,36 +690,132 @@ public function setBinaryBitsetBe($bitset)
 
     /**
      * Get a bit at the given ordinal number
-     * 
-     * @param $ordinal int Ordinal number of bit to get
+     *
+     * @param int $ordinal Ordinal number of bit to get
+     * @return boolean
+     */
+    public function getBit($ordinal)
```
Comment (prolic) [inline src/EnumSet.php]: again, check for $ordinal being integer?

**Comment by marc-mabe** [comment]
Did a benchmark (see #87) and the changes here doesn't look good:

```
this branch
+----------------+-----------------+--------+--------+------+-----+------------+---------+---------+---------+---------+---------+--------+----------+
| benchmark      | subject         | groups | params | revs | its | mem_peak   | best    | mean    | mode    | worst   | stdev   | rstdev | diff     |
+----------------+-----------------+--------+--------+------+-----+------------+---------+---------+---------+---------+---------+--------+----------+
| EnumSet66Bench | benchCountFull  |        | []     | 5000 | 50  | 1,089,824b | 0.592μs | 0.608μs | 0.606μs | 0.637μs | 0.011μs | 1.85%  | 0.00%    |
| EnumSet66Bench | benchCountEmpty |        | []     | 5000 | 50  | 1,089,824b | 0.592μs | 0.610μs | 0.606μs | 0.637μs | 0.012μs | 2.02%  | +0.26%   |
| EnumSet32Bench | benchCountFull  |        | []     | 5000 | 50  | 1,082,088b | 1.360μs | 1.395μs | 1.389μs | 1.462μs | 0.022μs | 1.59%  | +129.33% |
| EnumSet32Bench | benchCountEmpty |        | []     | 5000 | 50  | 1,082,088b | 1.359μs | 1.390μs | 1.384μs | 1.450μs | 0.017μs | 1.20%  | +128.49% |
+----------------+-----------------+--------+--------+------+-----+------------+---------+---------+---------+---------+---------+--------+----------+
```

```
current master
+----------------+-----------------+--------+--------+------+-----+------------+---------+---------+---------+---------+---------+--------+---------+
| benchmark      | subject         | groups | params | revs | its | mem_peak   | best    | mean    | mode    | worst   | stdev   | rstdev | diff    |
+----------------+-----------------+--------+--------+------+-----+------------+---------+---------+---------+---------+---------+--------+---------+
| EnumSet66Bench | benchCountFull  |        | []     | 5000 | 50  | 1,070,120b | 0.495μs | 0.511μs | 0.506μs | 0.536μs | 0.010μs | 2.02%  | +53.89% |
| EnumSet66Bench | benchCountEmpty |        | []     | 5000 | 50  | 1,070,120b | 0.497μs | 0.517μs | 0.506μs | 0.540μs | 0.012μs | 2.25%  | +55.76% |
| EnumSet32Bench | benchCountFull  |        | []     | 5000 | 50  | 1,062,384b | 0.321μs | 0.332μs | 0.329μs | 0.346μs | 0.007μs | 2.06%  | 0.00%   |
| EnumSet32Bench | benchCountEmpty |        | []     | 5000 | 50  | 1,062,384b | 0.321μs | 0.335μs | 0.330μs | 0.352μs | 0.009μs | 2.67%  | +1.02%  |
+----------------+-----------------+--------+--------+------+-----+------------+---------+---------+---------+---------+---------+--------+---------+
```

:(

**Comment by marc-mabe** [comment]
arg nothing to worry - I had a bug in my benchmarks - the PR is fixed now

**Comment by marc-mabe** [comment]
@prolic The PR is ready :)

File: src/EnumSet.php (hunk 2 above)
Comment (marc-mabe) [reply src/EnumSet.php]: It's using the same behavior as standard PHP function in PHP-5.6 and even with PHP-7.0 with strict-types=false.
-> We should discuss it in a different thread/issue

File: tests/MabeEnumTest/TestAsset/Enum31.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace MabeEnumTest\TestAsset;
+
+use MabeEnum\Enum;
+
+/**
+ * Unit tests for the class MabeEnum\Enum
+ *
+ * @link http://github.com/marc-mabe/php-enum for the canonical source repository
+ * @copyright Copyright (c) 2013 Marc Bennewitz
```
Comment (prolic) [inline tests/MabeEnumTest/TestAsset/Enum31.php]: It's 2017, maybe you should update that somewhen :)

Comment (marc-mabe) [reply tests/MabeEnumTest/TestAsset/Enum31.php]: good point -> but on another branch

**Pull request merged**